lib/puppet/indirector/catalog/compiler.rb

```diff
@@ -127,7 +127,33 @@ def inline_metadata(catalog, checksum_type)
       file = resource.to_ral
 
       if file.recurse?
-        # TODO: find and replace metadata for children recursively
+        file.recurse_remote_metadata do |meta|
+          # Don't create a new resource for the parent directory
+          next if meta.relative_path == "."
+
+          child_resource = Puppet::Resource.new(:file, File.join(file[:path], meta.relative_path))
+          child_resource[:ensure] = meta.ftype
+          # TODO: Make the coying of owner, group, and mode conditional on using permissions
+          child_resource[:owner]  = meta.owner.to_s
+          child_resource[:group]  = meta.group.to_s
+          child_resource[:mode]   = meta.mode.to_s(8).rjust(4, '0')
```

Comment:
Can we take the owner/group/mode lines out and implement that part in PUP-5697 along with the conditional logic? As it is now the static compiler will always cause the source permissions to be copied to the agent, which is the opposite of the non-static catalog behavior (when `source_permissions => ignore`):

```
Notice: /Stage[main]/Ntp/File[/tmp/dir/dir1]/group: group changed 'wheel' to 'staff'
Notice: /Stage[main]/Ntp/File[/tmp/dir/file]/group: group changed 'wheel' to 'staff'
Notice: /Stage[main]/Ntp/File[/tmp/dir/dir1/dir2]/group: group changed 'wheel' to 'staff'
Notice: /Stage[main]/Ntp/File[/tmp/dir/dir1/file1]/group: group changed 'wheel' to 'staff'
```